container/docker/handler.go

```diff
@@ -248,6 +251,7 @@ func newDockerContainerHandler(
 	handler.image = ctnr.Config.Image
 	handler.networkMode = ctnr.HostConfig.NetworkMode
 	handler.deviceID = ctnr.GraphDriver.Data["DeviceId"]
+	handler.restartCount = ctnr.RestartCount
```

Comment:
If I read this correctly, you are only updating handler.restartCount when a new docker container handler.  Have you verified that this increases when the container restarts?